mediaire_toolbox/transaction_db/model.py

```diff
@@ -98,6 +98,11 @@ def patient_consent(self):
         """
         return 1 if self.patient_consent_date is not None else 0
 
+    @patient_consent.expression
+    def patient_consent(self):
+        """To use in Query expressions, define SQL expression separately."""
+        return self.patient_consent_date != None  # noqa: E711
+
```

Comment:
I think this fixed it. SQLAlchemy was not able generate the correct SQL exporessions from the regular function definition, probably because `is not None` is not overloadable (I think). Anyway, you can provide a manual expression for the class case, that will be evaluated in `.filter()` statements, while the original definition will be used on the instance level, when actually accessing the field in a retrieved row.

Phew, that adds even more complexity, but it's actually a really useful feature. I'm glad I discovered it.